deepchem/data/pytorch_datasets.py

```diff
@@ -87,16 +88,25 @@ def __init__(self,
         self.deterministic = deterministic
         self.batch_size = batch_size
 
+    def __len__(self):
+        return len(self.disk_dataset)
+
     def __iter__(self):
         worker_info = torch.utils.data.get_worker_info()
         n_shards = self.disk_dataset.get_number_shards()
         if worker_info is None:
-            first_shard = 0
-            last_shard = n_shards
+            process_id = 0
```

Comment:
@arunppsg Can we add a unit test for the changed functionality?

Reply:
Documenting offline discussion, this is proving difficult to write. The existing unit tests pass. Will come back with additional tests in future PR